### ISSUE TITLE
Terraform error waiting for Lambda Function (Foobar-local) creation: unexpected state '', wanted target 'Active'. last error: %!s(<nil>)

### ISSUE THREAD

**Issue by jrr** [localstack/localstack#1835]
Terraform is failing to provision a lambda:
```
Error: error waiting for Lambda Function (FooBar-local) creation: unexpected state '', wanted target 'Active'. last error: %!s(<nil>)

  on ../foo-bar/foo-bar.tf line 26, in resource "aws_lambda_function" "example":
  26: resource "aws_lambda_function" "example" {
```

Here's the localstack log:

```
#!/bin/bash -eo pipefail
docker-compose up
Creating network "backend_default" with the default driver
Pulling localstack (localstack/localstack:)...
latest: Pulling from localstack/localstack

Creating backend_localstack_1_afedc2bf2864 ... =========================>]  452.3MB/452.3MBB
Creating backend_localstack_1_afedc2bf2864 ... doneAttaching to backend_localstack_1_c75c3ec62a73
localstack_1_c75c3ec62a73 | Waiting for all LocalStack services to be ready
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:29,576 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:29,582 INFO supervisord started with pid 14
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:30,586 INFO spawned: 'dashboard' with pid 20
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:30,588 INFO spawned: 'infra' with pid 21
localstack_1_c75c3ec62a73 | (. .venv/bin/activate; bin/localstack web)
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:30,598 INFO success: dashboard entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
localstack_1_c75c3ec62a73 | (. .venv/bin/activate; exec bin/localstack start --host)
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:31,600 INFO success: infra entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
localstack_1_c75c3ec62a73 | 2019-12-06T20:38:34:WARNING:bootstrap.py: Unable to load plugins from file /opt/code/localstack/.venv/lib/python3.7/site-packages/localstack_ext/plugins.py: No module named 'localstack_ext.utils.aws.aws_models'
localstack_1_c75c3ec62a73 | 2019-12-06T20:38:34:INFO:localstack.utils.common: Unable to store certificate file under /tmp/localstack/server.test.pem, using tmp file instead: [Errno 13] Permission denied: '/tmp/localstack/server.test.pem'
localstack_1_c75c3ec62a73 | Starting local dev environment. CTRL-C to quit.
localstack_1_c75c3ec62a73 | Starting mock S3 (http port 4572)...
localstack_1_c75c3ec62a73 | Starting mock IAM (http port 4593)...
localstack_1_c75c3ec62a73 | 2019-12-06T20:38:34:INFO:localstack.multiserver: Starting multi API server process on port 51492
localstack_1_c75c3ec62a73 | Starting mock Lambda service (http port 4574)...
localstack_1_c75c3ec62a73 | Starting mock CloudWatch Logs (http port 4586)...
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:35,980:API:  * Running on http://0.0.0.0:4557/ (Press CTRL+C to quit)
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:35,980:API:  * Running on http://0.0.0.0:4557/ (Press CTRL+C to quit)
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:35,980:API:  * Running on http://0.0.0.0:4557/ (Press CTRL+C to quit)
localstack_1_c75c3ec62a73 | 2019-12-06T20:38:35:DEBUG:localstack.services.awslambda.lambda_executors: Getting all lambda containers names.
localstack_1_c75c3ec62a73 | 2019-12-06T20:38:35:DEBUG:localstack.services.awslambda.lambda_executors: docker ps -a --filter="name=localstack_lambda_*" --format "{{.Names}}"
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:36,000:API:  * Running on http://0.0.0.0:4586/ (Press CTRL+C to quit)
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:36,000:API:  * Running on http://0.0.0.0:4586/ (Press CTRL+C to quit)
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:36,000:API:  * Running on http://0.0.0.0:4586/ (Press CTRL+C to quit)
localstack_1_c75c3ec62a73 | 2019-12-06T20:38:36:DEBUG:localstack.services.awslambda.lambda_executors: Removing 0 containers.
localstack_1_c75c3ec62a73 | 2019-12-06T20:38:36:INFO:localstack.services.awslambda.lambda_executors: Checking if there are idle containers.
localstack_1_c75c3ec62a73 | Waiting for all LocalStack services to be ready
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:41,105:API: 127.0.0.1 - - [06/Dec/2019 20:38:41] "GET / HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | Ready.
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,153:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "PUT /foo-bar-local-data HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,189:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "HEAD /foo-bar-local-data HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,257:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "HEAD /foo-bar-local-data HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,270:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?website= HTTP/1.1" 404 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,281:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "HEAD /foo-bar-local-data HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,300:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?website HTTP/1.1" 404 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,315:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?versioning= HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,322:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?accelerate= HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,334:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?requestPayment= HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,342:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?logging= HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,361:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?object-lock= HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,368:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?location= HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,376:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?tagging= HTTP/1.1" 404 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,382:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "HEAD /foo-bar-local-data HTTP/1.1" 200 -
localstack_1_c75c3ec62a73 | 2019-12-06 20:38:48,391:API: 127.0.0.1 - - [06/Dec/2019 20:38:48] "GET /foo-bar-local-data?website HTTP/1.1" 404 -
localstack_1_c75c3ec62a73 | 2019-12-06T20:38:48:DEBUG:localstack.services.awslambda.lambda_executors: Getting status for container "localstack_lambda_arn_aws_lambda_*********_000000000000_function_FooBar-local": docker ps -a --filter name='localstack_lambda_arn_aws_lambda_*********_000000000000_function_FooBar-local' --format "{{ .Status }} - {{ .Names }}" | grep -w "localstack_lambda_arn_aws_lambda_*********_000000000000_function_FooBar-local" | cat
Gracefully stopping... (press Ctrl+C again to force)
Stopping backend_localstack_1_c75c3ec62a73 ... 
Build was canceled
```

Here's my docker-compose:
```yaml
version: "2.1"

services:
  localstack:
    image: localstack/localstack
    ports:
      - "4500-4600:4500-4600"
      - "${PORT_WEB_UI-8080}:${PORT_WEB_UI-8080}"
    environment:
      - SERVICES=s3,iam,lambda
      # https://github.com/localstack/localstack/issues/1743#issuecomment-557942319
      - HOSTNAME=localhost
      - DEBUG=${DEBUG- }
      - DATA_DIR=${DATA_DIR- }
      - PORT_WEB_UI=${PORT_WEB_UI- }
      # - LAMBDA_EXECUTOR=${LAMBDA_EXECUTOR- }
      # https://github.com/localstack/localstack#configurations
      - LAMBDA_EXECUTOR=docker-reuse
      - KINESIS_ERROR_PROBABILITY=${KINESIS_ERROR_PROBABILITY- }
      - DOCKER_HOST=unix:///var/run/docker.sock
    volumes:
      - "${TMPDIR:-/tmp/localstack}:/tmp/localstack"
      - "/var/run/docker.sock:/var/run/docker.sock"
```

(this is run with `TMPDIR=~localstack` and `DEBUG=1`)

This is happening in CI (CircleCI's `ubuntu-1604:201903-01` [machine executor](https://circleci.com/docs/2.0/executor-types/#using-machine)).

What should I try next?

I have a couple of related questions:

- Is it possible to reference docker images anywhere in between (last release) and `latest`? More than once I've moved to `latest` for a bugfix, only to hit a regression a little bit later. Could [Docker Hub](https://hub.docker.com/r/localstack/localstack/tags) keep the intermediate versions around and available so that we can pin one?

- Would you be interested in a contribution of an example Terraform project to run in localstack's CI?

_Edit: changed the title after realizing that I initially wasn't waiting long enough for LocalStack to start up._

**Comment by jrr** [comment]
Found it! It's an incompatibility with recent versions of the Terraform AWS provider. Somehow I'd missed this in my issue searches:

https://github.com/localstack/localstack/issues/1818

https://github.com/terraform-providers/terraform-provider-aws/issues/11037

**Comment by whummer** [comment]
Thanks for reporting @jrr - looks like this could be a duplicate of #1822, should be fixed in #1838 . Can you please give it a try with the `latest` version of the Docker image? Thanks

**Comment by dineshviswanath** [comment]
@whummer I can confirm this issue has been resolved. even I have raised SO question here https://stackoverflow.com/questions/59224451/unable-to-create-lambdapermission-with-terraform-and-localstack-combination-cre/59228799#59228799
I have tried your fix and it working fine. thanks for putting fix! 👏 

Thanks for raising @jrr My goodness I was taking look at your well-documented issue at AWS provider repo. 🙌
